File: break_camel_case.js
function solution(string) {
  // Variables
  let result = [];

  for (i = 0; i < string.length; i++){              // For each character in the string...
    if (string[i] != string[i].toUpperCase()){      // if that character is not uppercase...
      result.push(string[i]);                       // push that character into the result array.
    }
    else {                                          // If that character is in fact uppercase...
      result.push(` ${string[i]}`);                 // push a space followed by that character into result
    }
  }
  return result.join("");                           // Return result joined together
}

File: break_camel_case.rb
def solution(string)
  # Variables
  result = []

  string.each_char do |x|       # For each character in the string...
    if x != x.upcase            # if that character is not uppercase...
      result << x               # shovel it into the result as is.
    elsif x == x.upcase         # if that character is in fact uppercase...
      result << " #{x}"         # shovel a space and then that character into the result.
    end
  end

  # Result
  result.join
end

# Practice writing solution in one line
def solution(string)
  string.chars.map { |s| s == s.downcase ? s : " #{s}" }.join
end
